assets/js/base/context/hooks/use-show-shipping-total-warning.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import { useDispatch, useSelect } from '@wordpress/data';
+import { CHECKOUT_STORE_KEY } from '@woocommerce/block-data';
+import { useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import {
+	areShippingMethodsMissing,
+	hasShippingRate,
+} from '@woocommerce/base-components/cart-checkout/totals/shipping/utils';
+
+/**
+ * Internal dependencies
+ */
+import { useShippingData } from './shipping/use-shipping-data';
+
+export const useShowShippingTotalWarning = () => {
+	const context = 'woocommerce/checkout-totals-block';
+	const errorNoticeId = 'wc-blocks-totals-shipping-warning';
+
+	const { shippingRates } = useShippingData();
+	const hasRates = hasShippingRate( shippingRates );
+	const prefersCollection = useSelect( ( select ) => {
+		return select( CHECKOUT_STORE_KEY ).prefersCollection();
+	} );
+	const { createInfoNotice, removeNotice } = useDispatch( 'core/notices' );
+	const shippingMethodsMissing = areShippingMethodsMissing(
+		hasRates,
+		prefersCollection,
+		shippingRates
+	);
+	useEffect( () => {
+		removeNotice( errorNoticeId, context );
+
+		if ( shippingMethodsMissing ) {
+			createInfoNotice(
+				__(
+					'Totals will be recalculated when a valid shipping method is selected.',
+					'woo-gutenberg-products-block'
+				),
+				{
+					id: 'wc-blocks-totals-shipping-warning',
+					isDismissible: false,
+					context,
+				}
+			);
+		}
+	}, [ shippingMethodsMissing, createInfoNotice, removeNotice ] );
```

Comment:
Please correct me if I'm wrong. In this part, don't we remove the notices every time this function is called independent whether they should be removed or not? If so, while I'm not a big fan of if-else-statements, wouldn't it be more efficient to remove the notices, only when they should be removed? I have something in mind like this:

```tsx
if ( shippingMethodsMissing ) {
  createInfoNotice( ... );
} else {
  removeNotice( ... );
}
```

Reply:
Good point Niels, I refactored this so it removes the notice if `shippingMethodsMissing` is false, and returns early. We don't need an `else`.